### PR TITLE
feat(handoff)!: drop <cli> positional from push/pull

### PR DESCRIPTION
**Pull request by kaiohenricunha**
## Summary

- Drops the redundant `<cli>` positional from `dotclaude-handoff push` / `pull`. The resolver already auto-detects across all three roots; forcing the user to state the source CLI was busywork.
- Adds `--from <cli>` to narrow explicitly (useful for scripting and resolving short-UUID collisions across roots).
- Adds `detectHost()` env-probe routing (`CLAUDECODE` / `CLAUDE_CODE_SSE_PORT` / `CODEX_*` / `COPILOT_*` / `GITHUB_COPILOT_*`) — all probes labelled UNCONFIRMED in the source. Bare `push` now narrows to the detected host's root and emits a one-line stderr fallback note (`host not detected, using latest across all clis: <short>` / `no current-session signal in claude, using latest claude session: <short>` / `using --from <cli> override, latest session: <short>`).
- Aligns zero-arg binary with SKILL.md: `dotclaude-handoff` with no args now executes `push` (was: print usage). Help lives behind `--help`.
- Breaking shim: `push|pull <cli> <query>` and bare `push|pull <cli>` exit 64 with a migration message pointing at `--from`. Power-user subs (`resolve`/`describe`/`digest`/`file`) keep their explicit `<cli> <id>`.

SKILL.md, the template copy, `docs/handoff-guide.md`, `docs/dotfile-quickstart.md`, and `CHANGELOG.md` are updated to match. SKILL.md now also flags the binary-vs-skill-runtime split for `--via github` explicitly.

## Test plan

- [x] `npm test` — 265/265 unit tests pass, including 9 new `detectHost` matrix tests (empty env, CLAUDECODE, SSE_PORT fallback, CODEX_\*, COPILOT_\*, priority ordering, negative prefix guards).
- [x] `bats plugins/dotclaude/tests/bats/dotclaude-handoff-five-form.bats` — 25/25 pass; includes 8 new/rewritten tests covering breaking shim (push + pull), `--from` narrowing (push + pull), `--from bogus` exits 64, `--from` beats `CLAUDECODE=1` override, stderr fallback notes.
- [x] `bats plugins/dotclaude/tests/bats/handoff-integration.bats` — 9/9 pass.
- [x] `bats plugins/dotclaude/tests/bats/handoff-concurrency.bats` — 5/5 pass.
- [x] Full bats suite across the repo — 223/223 pass.
- [x] `npm run lint` — clean (prettier + markdownlint + jsdoc coverage).
- [x] `npm run dogfood` — clean (skills manifest, specs, instruction drift, spec coverage).
- [x] Manual smokes pasted in the implementation thread:
  - No env → `host not detected, using latest across all clis: aaaa1111` → push succeeds.
  - `CLAUDECODE=1` → `no current-session signal in claude, using latest claude session: aaaa1111` → push succeeds.
  - `push claude aaaa1111` → exit 64 with the breaking-change message.
- [x] architect-reviewer pass: three P1 findings surfaced and applied (pull --from plumbed through, binary-vs-skill split documented, shim widened to catch `push claude` without third arg). Two P2s also applied (CLAUDECODE + --from override precedence test, prefix-only negative unit tests).

`npm run shellcheck` fails, but the failure pre-dates this branch (proved via `git stash`).

## No-spec rationale

This is a surgical refactor of a draft-maturity skill (`skills/handoff/SKILL.md` is `maturity: draft`), not a new feature area that warrants a spec doc. The change is scoped entirely to the `/handoff` public surface — no cross-cutting architectural decisions, no new protocols. Breaking-change semantics are captured in `CHANGELOG.md` under `[Unreleased] > BREAKING`; release-please will cut the minor bump from the `feat(handoff)!:` commit prefix. Migration guidance lives in the CHANGELOG entry and in the exit-64 error message the binary prints.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,33 +6,71 @@ All notable changes to `@dotclaude/dotclaude` land here. Format follows
...
+  for resolving short-UUID collisions across roots.
+- **`detectHost()` env-probe routing.** The binary best-effort
+  identifies the agentic CLI it is running inside via `CLAUDECODE`,
+  `CLAUDE_CODE_SSE_PORT`, and `CODEX_*` / `COPILOT_*` / `GITHUB_COPILOT_*`
+  prefix scans. All signals are labelled UNCONFIRMED in the source —
+  false positives are cheap (a narrower resolve) and false negatives
+  fall back to the union resolver.
+- **Honest stderr fallback notes.** Bare `push` (no query) now prints
+  one stderr line naming which fallback fired:
+  - `no current-signal in claude, using latest claude session: <short>`
```
[the remark this answers is not in the document]
Comment (kaiohenricunha) [reply CHANGELOG.md]: Fixed in 222f116 — CHANGELOG line now reads `no current-session signal in <cli>, using latest <cli> session: <short>` to match the actual binary output and use the `<cli>` placeholder generically.

**Comment by kaiohenricunha** [comment]
Test plan verified against HEAD 222f11640e50505f3e0d734450c526fa036778f4:
- `npm test` — ✓ local (265/265)
- `bats plugins/dotclaude/tests/bats/dotclaude-handoff-five-form.bats` — ✓ local (25/25)
- `bats plugins/dotclaude/tests/bats/handoff-integration.bats` — ✓ local (9/9)
- `bats plugins/dotclaude/tests/bats/handoff-concurrency.bats` — ✓ local (5/5)
- Full bats suite — ✓ local (223/223)
- `npm run lint` — ✓ local
- `npm run dogfood` — ✓ local
- Manual smokes (no-env / CLAUDECODE=1 / push claude aaaa1111) — ✓ local

Post-open cleanup applied: /simplify pass (commit ec3dcf0) removed dead `detectHostSession()`, collapsed inline resolver in power-user subs onto `resolveNarrowed()`, extracted `requireGitFallbackTransport` helper, swapped `Object.keys` for `for...in` in `detectHost`, skipped `enrichWithDescriptions` on bare pull (N unnecessary clones), and trimmed stale version narration. CHANGELOG stderr wording fix applied in 222f116 per Copilot inline review. Security review: clean (one INFO note on `for...in` prototype-chain reach, not exploitable).
